CHANGELOG.md

```diff
@@ -5,7 +5,7 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
-## [Unreleased]
+## [0.10] - 2024-05-31
 
 ### Added
```

Comment:
I looked through this [list of pull requests](https://github.com/SciTools-incubator/iris-esmf-regrid/pulls?page=1&q=is%3Apr+is%3Amerged+closed%3A%3E2023-11-01). Please could you add a reference to #319?